server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -487,7 +488,7 @@ public void testMatchWithOperator() {
     public void testMultiMatchWithOperator() {
         assertThat(
             convert("match((tags, name), 'foo bar') using best_fields with (operator='and')").toString(),
-            is("(name:foo bar | (+tags:foo +tags:bar))")
+            anyOf(is("(name:foo bar | (+tags:foo +tags:bar))"), is("((+tags:foo +tags:bar) | name:foo bar)"))
```

Comment:
I believe the order should not matter for a multi-match query.

Reply:
was this affected by your change, or was it flaky anyway?

Reply:
@matriv sorry for the late response. Yes, I re-ran and confirmed that the change to `MapFunction#evaluate` is the cause of this flakyness. 
On master, the same test update `anyOf(is("(name:foo bar | (+tags:foo +tags:bar))"), is("((+tags:foo +tags:bar) | name:foo bar)"))` was already made by me during Lucene 9.0 upgrade.

Reply:
Thank you!